blockchain/processing.py

```diff
@@ -429,6 +429,7 @@ def _execute_phase_3(self, config, phase_2_info):
             # storing valid verification record
             verification_db.insert_verification(phase_2_record)
 
+            # retrieve all phase 2 records for current block
             phase_2_records = self.get_sig_records(phase_2_record)
```

Comment:
@Voxelot and @j0j0r0 I think we need to keep this where it is. This is where we retrieve the records to check that the diversity requirements have been met.